### PR TITLE
Add the Code of Conduct

### PR DESCRIPTION
**Pull request by praneesha**
## Purpose
This adds the Code of Conduct to the B.io and updates it based on the latest version 2.0 as per [1].

[1] https://www.contributor-covenant.org/version/2/0/code_of_conduct

Fixes https://github.com/ballerina-platform/ballerina-dev-website/issues/129

## Check List

- [ ] **Page Addition**
  - [ ] Add `permalink` to pages
  - [ ] If contains empty folder(s), Add front-matter `redirect_to:`

- [ ] **Page Rename**
  - [ ] Add front-matter `redirect_from`
  - [ ] Add front-matter `redirect_to:` (If applicable)

### CODE REVIEW

File: code-of-conduct.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+</table>
+
+**2. Warning**
+
+<table border=1>
+<th> Community Impact </th>
+<th> Consequence </th>
+<tr>
+<td width=50%> A violation through a single incident or series of actions. </td>
+<td width=50%>> A warning with consequences for continued behavior. No interaction with the people involved including unsolicited interaction with those enforcing the code of conduct for a specified period of time. This includes avoiding interactions in community spaces as well as external channels like social media. Violating these terms may lead to a temporary or permanent ban.
```
Comment (anupama-pathirage) [inline code-of-conduct.md]:
```suggestion
<td width=50%> A warning with consequences for continued behavior. No interaction with the people involved including unsolicited interaction with those enforcing the code of conduct for a specified period of time. This includes avoiding interactions in community spaces as well as external channels like social media. Violating these terms may lead to a temporary or permanent ban.
```

File: code-of-conduct.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+</tr>
+</table>
+
+**3. Temporary ban**
+
+<table border=1>
+<th> Community Impact </th>
+<th> Consequence </th>
+<tr>>
+<td width=50%>> A serious violation of community standards including sustained inappropriate behavior. </td>
```
Comment (anupama-pathirage) [inline code-of-conduct.md]:
```suggestion
<td width=50%> A serious violation of community standards including sustained inappropriate behavior. </td>
```

File: code-of-conduct.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+<td width=50%>> A temporary ban from any sort of interaction or public communication with the community for a specified period of time. No public or private interaction with the people involved including unsolicited interaction with those enforcing the code of conduct is allowed during this period. Violating these terms may lead to a permanent ban.
```
Comment (anupama-pathirage) [inline code-of-conduct.md]:
```suggestion
<td width=50%> A temporary ban from any sort of interaction or public communication with the community for a specified period of time. No public or private interaction with the people involved including unsolicited interaction with those enforcing the code of conduct is allowed during this period. Violating these terms may lead to a permanent ban.
```

Comment (anupama-pathirage) [reply code-of-conduct.md]: Check all such places for incorrect formatting

File: code-of-conduct.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+<tr>>
+<td width=50%> Demonstrating a pattern of violation of community standards including sustained inappropriate behavior, harassment of an individual, or aggression towards or disparagement of classes of individuals. </td>
+<td width=50%> A permanent ban from any sort of public interaction within the project community.
+</td>
+</tr>
+</table>
+
+## Attribution
+
+This code of conduct is adapted from the [Contributor Covenant, version 2.0](#https://www.contributor-covenant.org/version/2/0/code_of_conduct).
```
Comment (anupama-pathirage) [inline code-of-conduct.md]: Link is not working

File: code-of-conduct.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+## Scope
+
+This Code of Conduct applies both within project spaces and in public spaces, and also applies when an individual is officially representing the project or its community in public spaces. Examples of representing our community include using an official e-mail address, posting via an official social media account, or acting as an appointed representative at an online or offline event.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be reported to the project team responsible for enforcement at **legal@wso2.com**. All complaints will be reviewed and investigated promptly and fairly.
+
+All project/community leaders are obligated to respect the privacy and security of the reporter of any incident.
```
Comment (anupama-pathirage) [inline code-of-conduct.md]: In some places, it says `project leaders`, some places `community leaders` and some places both. Shouldn't we use same terminology in all the places?

File: code-of-conduct.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
+---
+layout: ballerina-inner-page
+title: The Ballerina Code of Conduct
+permalink: /code-of-conduct/
+---
+
+# Contributor Covenant Code of Conduct
+
+This code of conduct aims at establishing a baseline standard of behavior to provide a safe and confortable environment for all members of the Ballerina community who are originated from different backgrounds.
```
Comment (anupama-pathirage) [inline code-of-conduct.md]:
```suggestion
This code of conduct aims at establishing a baseline standard of behavior to provide a safe and comfortable environment for all members of the Ballerina community who are originated from different backgrounds.
```

File: code-of-conduct.md (hunk 4 above)
Comment (praneesha) [reply code-of-conduct.md]: Removed the # sign to fix it.

File: code-of-conduct.md (hunk 5 above)
Comment (praneesha) [reply code-of-conduct.md]: Changed both occurrences to "project/community leaders".

File: code-of-conduct.md (hunk 2 above)
Comment (anupama-pathirage) [reply code-of-conduct.md]: Check this in all the places. Seems not fixed in all the places

File: code-of-conduct.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+Project/Community leaders will follow the below Community Impact Guidelines in determining the consequences for any action they deem in violation of this code of conduct:
+
+**1. Correction**
+
+<table border=1>
+<th> Community Impact </th>
+<th> Consequence </th>
+<tr border=1>
+<td width=50% border=1> Use of inappropriate language or other behavior deemed unprofessional or unwelcome in the community. </td>
+<td width=50% border=1> A private, written warning from community leaders providing clarity around the nature of the violation and an explanation of why the behavior was inappropriate. A public apology may be requested.
```
Comment (anupama-pathirage) [inline code-of-conduct.md]: Should we change this also to project/community leaders?

File: code-of-conduct.md (hunk 5 above)
Comment (anupama-pathirage) [reply code-of-conduct.md]: There is one more place. Added a comment.

File: code-of-conduct.md (hunk 2 above)
Comment (praneesha) [reply code-of-conduct.md]: Fixed.

File: code-of-conduct.md (hunk 3 above)
Comment (praneesha) [reply code-of-conduct.md]: Fixed.

File: code-of-conduct.md (hunk 7 above)
Comment (praneesha) [reply code-of-conduct.md]: Fixed.

**Pull request merged**